lib/improver/utilities/cube_manipulation_new.py

```diff
@@ -158,10 +97,179 @@ def strip_var_names(cubes):
+            coords_to_slice_over (list):
+                Dimension coordinates to slice over before concatenation.
+        """
+        self.master_coord = master_coord
+        self.coords_to_associate = coords_to_associate
+        self.coords_to_slice_over = coords_to_slice_over
+
+        if self.coords_to_slice_over is None:
+            self.coords_to_slice_over = ["realization", "time"]
+        if self.coords_to_associate is None and self.master_coord == "time":
+            self.coords_to_associate = ["forecast_period"]
+
+        # List of attributes to remove silently if unmatched
+        self.silent_attributes = ["history", "title", "mosg__grid_version"]
+
+    def _associate_any_coordinate_with_master_coordinate(self, cube):
+        """
+        Function to convert the given coordinates from scalar coordinates to
+        auxiliary coordinates, where these auxiliary coordinates will be
+        associated with the master coordinate.
+
+        For example, forecast_period can be converted from scalar coordinate
+        to auxiliary coordinate to be associated with a time dimension.
+
+        Args:
+            cube (iris.cube.Cube):
+                Cube requiring addition of the specified coordinates as
+                auxiliary coordinates.
+
+        Returns:
+            cube (iris.cube.Cube):
+                Cube where the the requested coordinates have been added to the
+                cube as auxiliary coordinates and associated with the desired
+                master coordinate.
+
+        Raises:
+            ValueError: If the master coordinate is not present on the cube.
+        """
+        coordinates = self.coords_to_associate
+        if coordinates is None:
+            coordinates = []
+
+        # If the master_coord is not a dimension coordinate, then the other
+        # coordinates cannot be associated with it.
+        if len(cube.coords(self.master_coord, dim_coords=True)) > 0:
+            for coord in coordinates:
+                if cube.coords(coord):
+                    temp_coord = cube.coord(coord)
+                    cube.remove_coord(coord)
+                    temp_aux_coord = (
+                        build_coordinate(temp_coord.points,
+                                         bounds=temp_coord.bounds,
+                                         coord_type=AuxCoord,
+                                         template_coord=temp_coord))
+                    coord_names = [
+                        coord.standard_name for coord in cube.dim_coords]
+                    cube.add_aux_coord(
+                        temp_aux_coord,
+                        data_dims=coord_names.index(self.master_coord))
+        return cube
+
+    @staticmethod
+    def _slice_over_coordinate(cubes, coord_to_slice_over):
+        """
+        Function slices over the requested coordinate and promotes the sliced
+        coordinate into a one-point dimension to help concatenation.
```

Comment:
This could do with a bit more detail, e.g.

> Function slices over the requested coordinate in each cube within a cubelist. The sliced coordinate is promoted into a one-point dimension to help concatenation. If the coord_to_slice_over is not found on a cube it is retained in the list in its original form.